### PR TITLE
Bug fixes for path and missing dependency errors

### PR DESCRIPTION
**Pull request by ITBlogger**
Fixes bugs related to explicit path missing from mkdir and test as well as dependency error from requiring File check that doesn't already exist in catalog.

### CODE REVIEW

**Comment by mhaskel** [comment]
@ITBlogger apologies it took so long to get to this, and thanks for the contribution! I'm taking a look now and will have some inline comments shortly. If you're still interested in getting this functionality added to the module, just address the comments and rebase, and we should be good to go :)

File: manifests/logical_volume.pp [hunk 1]
```diff
@@ -6,10 +6,17 @@
   $fs_type           = 'ext4',
   $mountpath         = "/${name}",
   $mountpath_require = false,
+  # pass needs to be a parameter for managing root (/) volume
```
Comment (mhaskel) [inline manifests/logical_volume.pp]: Please don't put comments inline in the parameter lists.

File: manifests/logical_volume.pp [hunk 2]
```diff
@@ -6,10 +6,17 @@
...
+  $pass              = '2'
 ) {
   validate_bool($mountpath_require)
 
   if $mountpath_require {
+    # Without this file mountpath resource, catalog runs fail for us on CentOS 6.5
+    file { "$mountpath":
+      ensure => directory,
+    }
```
Comment (mhaskel) [inline manifests/logical_volume.pp]: `$mountpath` should not be managed in the module. You should manage that in your node definition/role/profile.

File: manifests/logical_volume.pp [hunk 3]
```diff
@@ -40,6 +47,11 @@
     ensure  => $ensure,
     fs_type => $fs_type,
   }
+   
+  # Required, at least on our CentOS 6.5 build or Puppet errors out when trying to run mkdir or test
+  Exec {
+    path => '/bin:/sbin:/usr/bin:/usr/sbin',
+  }
```
Comment (mhaskel) [inline manifests/logical_volume.pp]: This is no longer needed.

**Comment by esalberg** [comment]
I spun off the pass and dump modifications to a separate pull request for clarity: https://github.com/puppetlabs/puppetlabs-lvm/pull/94

**Comment by mhaskel** [comment]
Thanks @ITBlogger @esalberg, fixed in #94

**Pull request closed without merging**